### PR TITLE
build: update `rest-model` module for spotbug 4.3.0 and re-enable plugin

### PR DESCRIPTION
**Pull request by mjsax**
Follow up to #7808

### CODE REVIEW

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/SessionProperties.java [hunk 1]
```diff
@@ -89,7 +92,11 @@ public KsqlHostInfo getKsqlHostInfo() {
   }
 
   public URL getLocalUrl() {
-    return localUrl;
+    try {
+      return new URL(localUrl.toString());
+    } catch (final MalformedURLException fatalError) {
+      throw new IllegalStateException("Could not deep copy URL: " + localUrl);
+    }
```
Comment (mjsax) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/SessionProperties.java:99]: This is a nasty one. Not straightforward to make a deep-copy (cf https://github.com/confluentinc/ksql/pull/7812/files#r671882647)

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/ConnectorList.java [hunk 2]
```diff
@@ -39,6 +40,7 @@ public ConnectorList(
     this.connectors = ImmutableList.copyOf(Objects.requireNonNull(connectors, "connectors"));
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "connectors is ImmutableList")
   public ImmutableList<SimpleConnectorInfo> getConnectors() {
```
Comment (mjsax) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/ConnectorList.java:44]: Compare https://github.com/confluentinc/ksql/pull/7809/files#r671807853

Or should we change the return type to `List`?

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/KsqlEntity.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ public String getStatementText() {
   }
 
   public List<KsqlWarning> getWarnings() {
-    return warnings;
+    return Collections.unmodifiableList(warnings);
```
Comment (mjsax) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/KsqlEntity.java:75]: Cannot use `ImmutableList` because `warnings` is build incrementally.

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/TypeList.java [hunk 4]
```diff
@@ -20,14 +20,15 @@
...
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import java.util.Map;
 import java.util.Objects;
 
 @Immutable
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class TypeList extends KsqlEntity {
 
-  private final ImmutableMap<String, SchemaInfo> types;
+  private final Map<String, SchemaInfo> types;
```
Comment (jzaralim) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/TypeList.java:31]: Why was this changed to `Map`?

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/QueryDescription.java [hunk 5]
```diff
@@ -101,6 +102,7 @@ public String getStatementText() {
     return windowType;
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "fields is ImmutableList")
   public List<FieldInfo> getFields() {
```
Comment (jzaralim) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/QueryDescription.java:106]: This goes for a lot of other files as well, but should we change the internal representation to `ImmutableList`?

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/TypeList.java (hunk 4 above)
Comment (mjsax) [reply ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/TypeList.java]: Not 100%, but it broke the test. (Will revert.)

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/ApiJsonMapper.java [hunk 6]
```diff
@@ -61,6 +62,7 @@
           .addSerializer(Date.class, new DateSerializer())
       );
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "should be mutable")
```
Comment (mjsax) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/ApiJsonMapper.java:65]: Must be mutable, otherwise `StreamedRowTest` breaks (not 100% sure though).

**Review by agavra — APPROVED** [review]
